src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -192,6 +198,10 @@ public AnomalyDetector(
         this.categoryFields = categoryFields;
         this.user = user;
         this.detectorType = isMultientityDetector(categoryFields) ? MULTI_ENTITY.name() : SINGLE_ENTITY.name();
+        this.resultIndex = Strings.trimToNull(resultIndex);
+        if (this.resultIndex != null && !this.resultIndex.startsWith(CUSTOM_RESULT_INDEX_PREFIX)) {
+            throw new IllegalArgumentException(INVALID_RESULT_INDEX_PREFIX);
+        }
```

Comment:
@amitgalitz @ylwu-amzn will this custom-result-index check make it into the initial validation API?

Reply:
I don't have this currently as part of the backend. We can discuss offline if it is practical to open another PR to add this validation check but not sure if there is time for 1.2

Reply:
Seems not so big change to add this to validation API, but I'm ok to leave it to 1.3 if no time.

Reply:
If this is the only "blocker" check happening then I could definitely add this in after PR's are merged as code currently doesn't have these variables and etc.